tailcfg/tailcfg.go

```diff
@@ -1223,6 +1223,26 @@ const (
 // the application via the WhoIs API.
 type PeerCapMap map[PeerCapability][]json.RawMessage
 
+// GetCap returns the values of the given capability JSON unmarshaled into the
+// provided type. It returns nil, nil if the capability is not present. It
+// returns an error if the capability is present but the value is not of the
+// provided type.
+func GetCap[T any](cm PeerCapMap, cap PeerCapability) ([]*T, error) {
```

Comment:
Instead of GetCap how about UnmarshalCapJSON? Gets JSON in there and unmarshal so you know the error is related to that. Other Get returning no error on miss is kinda weird?

Reply:
`UnmarshalCapJSON` sgtm

Reply:
Does it need to be in tailcfg? This tries to be a leaf data type package without much code. This is fine but worried about additions later with new deps

Reply:
I wanted to keep it closer to `PeerCapMap`, I agree with concern about this turning into a data package though `encoding/json` is not new so I thought it was ok.

Reply:
Done, PTAL at docs